variables.tf

```diff
@@ -0,0 +1,206 @@
+variable "public_ip_config" {
+  type = object({
+      enabled           = bool
+      allocation_method = optional(string, "Static")
+  })
+  default = {
+    enabled = false
+  }
+  validation {
+    condition     =  contains(["Static","Dynamic"], var.public_ip_config.allocation_method)
+    error_message = "Allocation method must be Static or Dynamic"
+  }
+  description = <<-DOC
+  ```
+    enabled: Optionally select true if a public ip should be created. Defaults to false.
+    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
+  ```
+  DOC
+} 
+
+# nsg needs to be an object to use the count object in main.tf. 
+variable "nic_config" {
+  type = object({
+    private_ip  = optional(string)
+    dns_servers = optional(list(string))
+    nsg = optional(object({
+      id = string
+    }))
+  })
+  default     = {}
+  description = <<-DOC
+  ```
+    private_ip: Optioanlly specify a private ip to use. Otherwise it will  be allocated dynamically.
+    dns_servers: Optionally specify a list of dns servers for the nic.
+    nsg_id: Optinally specify the id of a network security group that will be assigned to the nic.    
+  ```
+  DOC
+}
+
+variable "enable_accelerated_networking" {
+  description = "Enabled Accelerated networking (SR-IOV) on the NIC. The machine SKU must support this feature. https://docs.microsoft.com/en-us/azure/virtual-network/create-vm-accelerated-networking-powershell"
+  type        = bool
+  default     = "false"
+}
+
+variable "additional_network_interface_ids" {
+  type        = list(string)
+  default     = []
+  description = "List of ids for additional azurerm_network_interface."
+}
+
+variable "subnet" {
+  type               = object ({
+    id               = string
+    address_prefixes = list(string)
+  })
+  description = "The variable takes the subnet as input and takes the id and the address prefix for further configuration."
+}
+
+variable "virtual_machine_config" {
+  type = object({
+      hostname                     = string
+      size                         = string
+      location                     = string
+      admin_username               = optional(string, "loc_sysadmin")
+      os_sku                       = optional(string, "22_04-lts-gen2")
+      os_offer                     = optional(string, "0001-com-ubuntu-server-jammy")
+      os_version                   = optional(string, "latest")
+      os_publisher                 = optional(string, "Canonical")
+      os_disk_caching              = optional(string, "ReadWrite")
+      os_disk_size_gb              = optional(number, 64)
+      os_disk_storage_type         = optional(string, "StandardSSD_LRS")
```

Comment:
I think we should change the default to premium_lrs

Reply:
Fixed as recommend.